wordpress-mobile/WordPress-Android
# Plans in site creation: Plan selection screen

Fixes #19284 

| Domains | Plans (to be updated) |
|--------|--------|
| ![Screenshot_20230930_121051](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/84e50c49-1757-411d-8963-a978cc6b9b4e) | ![Screenshot_20231015_083133](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/fab8085d-33b1-498f-a187-d600c19052af)|

To test:

>**Note**
> Waiting for https://github.com/Automattic/wp-calypso/pull/82135

- `Launch` JP app
- `Go` to Me-> Debug settings
- `Switch` to `Remote Feature` tab, find `plans_in_site_creation` and enable it
-  `Switch` to **My Site** tab
- `Tap` the down arrow to the right of the site name on the My Site screen
- `Tap` the + icon at the top right on the Site picker screen
- `Select` ⊕ **Create WordPress.com site** on the modal dialog
- `Tap` Skip at top right on the first step (topic screen)
- `Tap` Skip at top right on the next step (theme screen)
- `Enter` a search query (whatever you want here)
- `Verify` that result domains are as shown in the screenshot above
- `Select` a domain
- `Verify` Plans screen opens with Personal, Premium, Business, and Ecommerce plans
- `Select` a plan 
- `Verify` it return to site setup screen 
- `Verify` it opens checkout, after site setup when an annual paid plan is chosen
- `Verify` it doesn't open checkout if a free domain with free plan is chosen, and site setup is completed
- `Verify` it doesn't allow a annual plan to be selected with a free domain
- Also, `Verify`, select a paid domain on the previous step, then choose to continue with a free domain, free plan

Checkout
- Setup sandbox cookie to test checkout (see instructions on Test Plan)
- Complete checkout with a paid domain, and paid annual plan


## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested various flows

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:
